speasy/amda/amda.py

```diff
@@ -95,6 +95,11 @@ def __init__(self, wsdl: str = 'AMDA/public/wsdl/Methods_AMDA.wsdl', server_url:
         self.catalog = {}
         if "AMDA/inventory" in _cache:
             self._unpack_inventory(_cache["AMDA/inventory"])
+            if any(map(lambda d: d == {},
+                       [self.parameter, self.mission, self.observatory, self.instrument, self.dataset,
+                        self.datasetGroup, self.component, self.dataCenter, self.folder, self.timeTable,
+                        self.catalog])):
+                self.update_inventory()
```

Comment:
Not sure it worth it to write a specific test since this line is almost unreachable